assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -612,21 +645,19 @@ angular.module('openshiftConsole')
       }
       triggers = triggers.concat(_.filter($scope.buildConfig.spec.triggers, function(obj) { return obj.type === "Generic" }));
 
-      if ($scope.triggers.imageChange) {
-        var imageChangeTriggers = _.filter($scope.buildConfig.spec.triggers, function(obj) { return obj.type === "ImageChange" });
-        if (imageChangeTriggers.length === 0) {
-          imageChangeTriggers.push({
-            imageChange: {},
-            type: "ImageChange"
-          });
-        }
-        triggers = triggers.concat(imageChangeTriggers);
-      }
-      if ($scope.triggers.configChange) {
+      if (presentTriggers.configChange) {
         triggers.push({
           type: "ConfigChange"
         });
       }
+
+      // Add all imageChange triggers to the triggers array. If builder imageChange trigger is checked, keep the trigger,
+      // otherwise delete it.
```

Comment:
It feels a little odd to always add a builder image trigger and then remove it when the checkbox isn't checked, but it does make the logic easier here. Can you add a line to the comment saying the builder image change trigger is always added to `imageChangeTriggers` in `getTriggerMap`?